src/main/java/com/celements/webform/ActionService.java

```diff
@@ -8,49 +8,57 @@
 import org.slf4j.LoggerFactory;
 import org.xwiki.component.annotation.Component;
 import org.xwiki.component.annotation.Requirement;
+import org.xwiki.configuration.ConfigurationSource;
 import org.xwiki.context.Execution;
+import org.xwiki.model.reference.DocumentReference;
 import org.xwiki.velocity.VelocityManager;
 
-import com.celements.web.service.IWebUtilsService;
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentNotExistsException;
 import com.xpn.xwiki.XWikiContext;
-import com.xpn.xwiki.XWikiException;
 import com.xpn.xwiki.api.Document;
 import com.xpn.xwiki.doc.XWikiDocument;
 import com.xpn.xwiki.web.Utils;
 
 @Component
 public class ActionService implements IActionServiceRole {
 
-  private static Logger _LOGGER = LoggerFactory.getLogger(ActionService.class);
+  private static Logger LOGGER = LoggerFactory.getLogger(ActionService.class);
 
   @Requirement
-  private IWebUtilsService webUtilsService;
+  IModelAccessFacade modelAccess;
+
+  @Requirement
+  private ConfigurationSource configSource;
 
   @Requirement
   private Execution execution;
 
   @Override
   public boolean executeAction(Document actionDoc, Map<String, String[]> request,
       XWikiDocument includingDoc, XWikiContext context) {
-    _LOGGER.info("Executing action on doc '" + actionDoc.getFullName() + "'");
+    LOGGER.info("Executing action on doc '" + actionDoc.getFullName() + "'");
     VelocityContext vcontext = getVelocityManager().getVelocityContext();
     vcontext.put("theDoc", actionDoc);
-    Boolean debug = (Boolean) vcontext.get("debug");
-    vcontext.put("debug", true);
-    Boolean hasedit = (Boolean) vcontext.get("hasedit");
-    vcontext.put("hasedit", true);
+    Object vDebugBefore = vcontext.get("debug");
+    final Object configScriptDebug = configSource.getProperty("actionScriptDebug");
```

Comment:
naming: config source properties we normally head with project and component name e.g. 
```suggestion
    final Object configScriptDebug = configSource.getProperty("celements.action.scriptDebug");
```